Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+
+**Example:**
+```
+let filtered = [112, 52, 0, -1, 944].filter(value => value>0);
+console.log(filtered);
+ ```
+*Output:*
+
+    [112,52,944]
+    
+### [3. reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)
+
+**Description:**
+reduces the array to a single value and executes provided function for each value of the array (from left-to-right).
+
+**Syntax:**
+ * _array.reduce( function(total, currentValue, currentIndex, arr), initialValue )_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| function(currentValue,index,arr) | Required function to be run for each array element |
+|initialValue | Optional.|
+
+**Return value:**
+returns accumulated result fron the last call of the callback function.
+
+**Example:**
+```
+const array1 = [1, 2, 3, 4];
+const reducer = (accumulator, currentValue) => accumulator + currentValue;
+console.log(array1.reduce(reducer));
+console.log(array1.reduce(reducer, 5));
+```
+*Output:*
+    
+    10
+    15
+
+### [4. slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)
+
+**Description:**
+Returns the selected elements in an array, as a new array object and selects the elements starting at the given start argument, and ends at the given end argument(excluding end argument).
+
+**Syntax:**
+ * _array.slice(start, end)_
+
+ | Parameter | Description |
+ | :-------- | :---------- |
+ | start     | Optional. specifies starting index of the selection    |
+ | end       | Optional. specifies ending index of the selection    |
+
+ **Return value:**
+ New array containing the selected elements.
+
+ **Example:**
+```
+    let arr = [23,56,87,32,75,13];
+    let new_arr = arr.slice();
+    console.log(arr);
+    console.log(new_arr);
+```    
+*Output:*
+
+    [23,56,87,32,75,13]
+    [23,56,87,32,75,13]
+
+### [5. splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
+
+**Description:**
+Adds/removes items to/from an array, and returns the removed item(s).
+
+**Syntax:**
+ * _array.splice(index, howmany, item1, ....., itemX)_
+
+| Parameter | Description|
+| :-------- | :--------- |
+| index	    | Required. An integer that specifies at what position to add/remove items |
+| howmany | Optional. The number of items to be removed.|
+| item1, ..., itemX | Optional. The new item(s) to be added to the array |
+
+**Return value:**
+Array without the removed items.
+
+**Example:**
+```
+const languages = ['C++', 'Java', 'Html', 'Python', 'C']; 
+console.log(languages); 
+// Add 'Julia' and 'Php' after removing 'Html'. 
+const removed = languages.splice(2, 1, 'Julia', 'Php') 
+console.log(languages); 
+console.log(removed); 
+```
+*Output:*
+
+    C++,Java,Html,Python,C
+    C++,Java,Julia,Php,Python,C
+
+### [6. concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)
+
+**Description:**
+used to join two or more arrays.

```

Comment:
It can also be used to add a single item (not an array) and that is how it is more commonly used so please focus your description and examples on that.

Please add a note comparing it to `.push` in that this does not mutate the array but creates a new one.